File: solutions/7.cpp
#include <iostream>
#include <vector>

using namespace std;

bool isPrime(int number) {
	int i = 2;
	while (i < number) {
		if (!(number % i)) {
			return false;
		}
		i++;
	}
	return true;
}

int main() {
	vector<int> primes;
	int index = 10001;
	int i = 2;
	while (primes.size() < index) {
		if (isPrime(i)) {
			cout << "prime[" << primes.size() << "] = " << i << endl;
			primes.push_back(i);
		}
		i++;
	}
	cout << primes[index - 1] << endl;
	return 0;
}

File: solutions/7.php
<?php
	function isPrime($num) {
		$i = 2;
		while ($i < $num) {
			if (!($num % $i)) {
				return false;
			}
			$i = $i + 1;
		}
		return true;
	}
	$index = 10001;
	$primes = array();
	$i = 2;
	while (count($primes) < 10001) {
		if (isPrime($i)) {
			$primes[] = $i;
			echo "primes[" . count($primes) - 1 . "] = " . $i . "\n";
		}
		$i = $i + 1;
	}
	echo $primes[$index - 1] . "\n";
?>